feat: add keybindings reference dialog to TUI

frankdierolf opened the pull request:
## Problem

When using the TUI, there's no easy way to discover what keybindings are available. Users have to check documentation or dig through config files to find out what shortcuts they can use.

## Solution

Added a searchable keybindings dialog accessible via `<leader>?` that displays all 88 keybindings with their descriptions.

## What Changed

- Created `dialog-keybindings.tsx` using the existing `DialogSelect` component
- Added `keybindings_list` config option (default: `<leader>?` = ctrl+x + ?)
- Registered in command palette under "System" category
- Descriptions dynamically extracted from config schema using `zodToJsonSchema`

## How to Use

**Via Keybinding:** Press `ctrl+x` then `?`  
**Via Command Palette:** Press `ctrl+p`, type "keybindings"

## Implementation

Follows existing patterns:
- Reuses `DialogSelect` component (same as model/theme lists)
- Fetches keybindings from `useKeybind()` hook
- Alphabetically sorted and searchable by default
- Descriptions stay in sync with schema automatically (single source of truth)

Fixes #3886

frankdierolf committed:
feat: add keybindings reference dialog to TUI

Added a searchable keybindings dialog accessible via <leader>?
that displays all 88 keybindings with their descriptions.

Features:
- Accessible via ctrl+x then ? or command palette
- Shows all keybindings in alphabetical order
- Searchable and filterable using DialogSelect
- Descriptions dynamically extracted from config schema

Implementation:
- Created dialog-keybindings.tsx component
- Added keybindings_list config option (default: <leader>?)
- Registered in command palette under "System" category
- Uses zodToJsonSchema to extract descriptions at runtime

This ensures descriptions stay in sync with schema definitions
automatically - single source of truth, zero maintenance.

Fixes #3886

frankdierolf commented:
### Hola, 

Please take some seconds to review. 
Thats a feautre which tangles the UX space as, thats why^^

- I throwed in under "System" which seemed reasonable to me: 
- And I made the decision for `<leader> + ?` 
- And I ordered the overview alphabetically. 
- Only READ
- Didn't test how it will behave with custom keybindings. Should be fine, but don't know. 
 
<img width="981" height="1076" alt="image" src="https://github.com/user-attachments/assets/654da4e7-3ae2-440f-9127-e16eca41d154" />

<img width="981" height="1076" alt="image" src="https://github.com/user-attachments/assets/94d7867c-b5fc-4092-ad26-1df61508c286" />

#### Best regrads,
Frank :v:

frankdierolf commented:
Updated to latest dev branch

agentic-ai-forge commented:
Tested this on Linux (Fedora 42, Bun 1.3.5) after rebasing onto latest `dev` - works great! The `<leader>?` shortcut opens a clean, searchable keybindings dialog.

This is a really useful feature for discoverability. Would love to see this merged. Thanks @frankdierolf!

frankdierolf commented:
@agentic-ai-forge Thanks for mentioning, forgot about that PR. Let me work that out.

frankdierolf commented:
Updated to latest dev + fixed some stuff:

- Rebased clean (no merge mess)
- Added all 89 keybindings now (was missing like 40)
- Fixed patterns: added useDialog hook, onSelect handler, memoized options
- Dialog now closeable with Escape/Enter

btw, I don't really understand why these 3 tests fail (patch/revert/diff with worktrees) - seems like a CI thing? Our changes only touch dialog-keybindings.tsx and auto-generated types. @thdxr you've been working on the snapshot/worktree stuff lately, maybe you can take a quick look and explain what's wrong? ^^

Ready for another look :v:

frankdierolf commented:
Hola, 

**First of all:**
Tests are passing now after rebasing onto latest dev.

The failures were because the branch was missing recent worktree infrastructure commits (0b4af9522 and 5944443a6 from yesterday). The snapshot tests need those to properly handle worktree contexts. Rebased cleanly since this PR only touches UI components - no conflicts.

**Second,**
if that keybinding help feature is desired it looks good to me, i just followed the already given patterns in the codebase, Technically not much concering at least to my eyes^^

Said that, the question remains, it involves UX. Currently the command pallete already shows shortcuts so that could be seens as a duplicate.  Depends on the taste of the opencode community. Personally, I still have to look up shortcuts ^-^

Thanks for the great work, its a joy to see the project grow in that fast pacing space :grin: 

Happy new year everyone and best regards,
Frank :v:

*Edit: Rushed thruh my head, the keybindings are currently static, thats maybe not optimal.

frankdierolf commented:
Fixed the static descriptions issue from my edit above ^^

Now using `zodToJsonSchema` to dynamically extract descriptions from the config schema - same pattern we use in server.ts. This way descriptions stay in sync automatically, single source of truth.

Also squashed everything into one clean commit and rebased onto latest dev for nice history.

antonio-ivanovski commented:
> Nice. I was just about to get this annoyance implemented myself but you beat me to the race few months ahead. Thanks! Hope it is merged it soon.
> <img alt="image" width="658" height="254" src="https://private-user-images.githubusercontent.com/1628876/538059089-132639a9-3bab-422f-8ccb-a718055a8be3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg5MjIwMDksIm5iZiI6MTc2ODkyMTcwOSwicGF0aCI6Ii8xNjI4ODc2LzUzODA1OTA4OS0xMzI2MzlhOS0zYmFiLTQyMmYtOGNjYi1hNzE4MDU1YThiZTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAxMjBUMTUwODI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTdjYWMxM2JiOTg5YjhlZDY2OWYyYmNhODZiODNlODVmNTFiZWY1OTBiYzlkNDdlMTI2ZTNmYTUxYjM4NDU2YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.CvI_bVrPzFpc8HmgRI29PgfWjspNYtNzBbAimcmSfrA">

As experiment, i let it implement it, it prompted me where to place it and whether it is just for the TUI or also for Desktop and Browser. GPT 5.2 (High) created this, tried GPT 5.2 Codex but it used most of its context trying to explore the project so i canceled it, Opus 4.5 was also good but decided to use the same worktree as GPT 5.2, it saw that it was implemented and it stopped.

<img width="627" height="393" alt="image" src="https://github.com/user-attachments/assets/97248d54-e4af-4b9a-9c67-73552bacabb2" />
